domain/status/service/service.go

```diff
@@ -302,14 +316,8 @@ func (s *LeadershipService) SetRelationStatus(
 	relationUUID corerelation.UUID,
 	info corestatus.StatusInfo,
 ) error {
-	// Get application name for leadership check.
```

Comment:
See the note about the flyby in the PR description, this was going into state to get the application name when it could just get it from the unit name.